lib/agent0/agent0/hyperdrive/policies/lpandarb.py

```diff
@@ -0,0 +1,642 @@
+    return total_shares_needed, total_bonds_needed
+
+
+# TODO this should maybe subclass from arbitrage policy, but perhaps making it swappable
+class LPandArb(HyperdrivePolicy):
+    """LP and Arbitrage in a fixed proportion."""
+
+    @classmethod
+    def description(cls) -> str:
+        """Describe the policy in a user friendly manner that allows newcomers to decide whether to use it.
+
+        Returns
+        -------
+        str
+        A description of the policy
+        """
+        raw_description = """
+        LP and arbitrage in a fixed proportion.
+        If no arb opportunity, that portion is LPed. In the future this could go into the yield source.
+        Try to redeem withdrawal shares right away.
+        Arbitrage logic is as follows:
+        - If the fixed rate is higher than `high_fixed_rate_thresh`:
+            - Close entire short and open a new long for `trade_amount` base.
+        - If the fixed rate is lower than `low_fixed_rate_thresh`:
+            - Close entire long and open a new short for `trade_amount` bonds.
+        """
+        return super().describe(raw_description)
+
+    @dataclass
+    class Config(HyperdrivePolicy.Config):
+        """Custom config arguments for this policy.
+
+        Attributes
+        ----------
+        high_fixed_rate_thresh: FixedPoint
+            The upper threshold of the fixed rate to open a position
+        low_fixed_rate_thresh: FixedPoint
+            The lower threshold of the fixed rate to open a position
+        lp_portion: FixedPoint
+            The portion of capital assigned to LP
+        """
+
+        lp_portion: FixedPoint = FixedPoint("0.5")
+        high_fixed_rate_thresh: FixedPoint = FixedPoint("0.1")
+        low_fixed_rate_thresh: FixedPoint = FixedPoint("0.02")
+        rate_slippage: FixedPoint = FixedPoint("0.01")
+
+        @property
+        def arb_portion(self) -> FixedPoint:
+            """The portion of capital assigned to arbitrage."""
+            return FixedPoint(1) - self.lp_portion
+
+    def __init__(
+        self,
+        budget: FixedPoint,
+        rng: NumpyGenerator | None = None,
+        slippage_tolerance: FixedPoint | None = None,
+        policy_config: Config | None = None,
+    ):
+        """Initialize the bot.
+
+        Arguments
+        ---------
+        budget: FixedPoint
+            The budget of this policy
+        rng: NumpyGenerator | None
+            Random number generator
+        slippage_tolerance: FixedPoint | None
+            Slippage tolerance of trades
+        policy_config: Config | None
+            The custom arguments for this policy
+        """
+        # Defaults
+        if policy_config is None:
+            policy_config = self.Config()
+        self.policy_config = policy_config
+        self.arb_amount = self.policy_config.arb_portion * budget
+        self.lp_amount = (self.policy_config.lp_portion) * budget
+        self.minimum_trade_amount = FixedPoint("10")
```

Comment:
FYI this does not need to be in quotes because it is an integer, so precision is preserved. Quotes are only needed for floats. NBD either way, just letting you know in case you didn't (this was not the case when we first launched fixedpointmath, but is now).

Reply:
Also, why are you hard coding this? The API pool config provides a contract-specified minimum_trade_amount. Is this value intentionally different? if so, why?

Reply:
thanks, been confused by this

Reply:
thanks, I didn't know about minimumTransactionAmount